tests/phpunit/tests/base-sender.php

```diff
@@ -3,38 +3,75 @@
 class Test_WPNotify_BaseSender extends WPNotify_TestCase {
 
 	/**
-	 * @param string $sender_name
-	 * @param string $expected_json
+	 * @param string|array $sender
+	 * @param string       $expected_json
 	 *
 	 * @dataProvider data_provider_senders
 	 */
-	public function test_it_can_be_json_encoded( $sender_name, $expected_json ) {
+	public function test_it_can_be_json_encoded( $sender, $expected_json ) {
 
-		$senderInstance = new WPNotify_BaseSender( $sender_name );
-		$encoded        = json_encode( $senderInstance );
-		$this->assertEquals( $expected_json, $encoded );
+		$sender_instance = null;
+
+		if ( is_string( $sender ) ) {
+			$sender_instance = new WPNotify_BaseSender( $sender );
+		} else if ( is_array( $sender ) ) {
+			$sender_instance = new WPNotify_BaseSender( $sender['name'], $sender['image'] );
+		}
```

Comment:
As a rule of thumb, I'd rather not have conditional logic in our tests: the less we have to think about what the test verifies, the quicker we could read them.

Maybe you could instantiate these objects in the `@dataProvider` instead. Or maybe you could use two different tests with no provider.

Reply:
fixed